bezier-rs/lib/src/lib.rs

```diff
@@ -853,6 +838,159 @@ impl Bezier {
+	/// - maximize_arcs - [MaximizeArcs] - The default value is `Automatic`.,
+	/// - error: f64 - The error used for approximating the arc's fit. The default is `0.5`.
+	/// - max_iterations: i32 - The maximum number of segment iterations used as attempts for arc approximations. The default is `100`.
+	pub fn arcs(&self, arcs_options: ArcsOptions) -> Vec<CircleArc> {
+		let ArcsOptions { maximize_arcs, error, max_iterations } = arcs_options;
+		match maximize_arcs {
+			MaximizeArcs::Automatic => {
+				let (auto_arcs, final_low_t) = self.arcs_helper(0., 1., error, max_iterations, true);
+				if final_low_t != 1. {
+					[
+						auto_arcs,
+						self.split(final_low_t)[1].arcs(ArcsOptions {
+							maximize_arcs: MaximizeArcs::Off,
+							error,
+							max_iterations,
+						}),
+					]
+					.concat()
+				} else {
+					auto_arcs
+				}
+			}
+			MaximizeArcs::On => self.arcs_helper(0., 1., error, max_iterations, false).0,
+			MaximizeArcs::Off => self
+				.get_extrema_t_list()
+				.windows(2)
+				.flat_map(|t_pair| self.arcs_helper(t_pair[0], t_pair[1], error, max_iterations, false).0)
+				.collect::<Vec<CircleArc>>(),
+		}
+	}
+
+	/// Implements an algorithm that approximates a bezier curve with circular arcs.
+	/// This algorithm uses a method akin to binary search to find an arc that approximates a maximal segment of the curve.
+	/// Once a maximal arc has been found for a sub-segment of the curve, the algorithm continues by starting again at the end of the previous approximation.
+	/// More details can be found in the [Approximating a Bezier curve with circular arcs](https://pomax.github.io/bezierinfo/#arcapproximation) section of Pomax's bezier curve primer.
+	/// A caveat with this algorithm is that it is possible to find erroneous approximations in cases such as in a very narrow `U`.
+	/// - `stop_when_invalid`: Used to determine whether the algorithm should end early erroneous approximations are encountered.
+	///
+	/// Returns a tuple where the first element is the list of circular arcs and the second is the `t` value where the next segment should start from.
+	/// The second value will be `1.` except for when `stop_when_invalid` is true and an invalid approximation is encountered.
+	fn arcs_helper(&self, local_low: f64, local_high: f64, error: f64, max_iterations: i32, stop_when_invalid: bool) -> (Vec<CircleArc>, f64) {
+		let mut low = local_low;
+		let mut middle = local_low + (local_high - local_low) / 2.;
+		let mut high = local_high;
+		let mut previous_high = local_high;
+
+		let mut iterations = 0;
+		let mut previous_arc = CircleArc::default();
+		let mut was_previous_good = false;
+		let mut arcs = Vec::new();
+
+		// Outer loop to iterate over the curve
+		while low < local_high {
+			if iterations > max_iterations {
+				break;
+			}
+			// Inner loop to find the next maximal segment of the curve that can be approximated with a circular arc
+			loop {
+				if iterations > max_iterations {
+					break;
+				}
```

Comment:
I think you can combine this as `while iterations <= max_iterations {`? And then that would let you remove the `if iterations > max_iterations { break; }` on the previous few lines? Double check my reasoning. As it currently stands, this feels a bit messy, and I think that would help clean it up.